applications/GeoMechanicsApplication/custom_conditions/T_normal_flux_condition.cpp

```diff
@@ -0,0 +1,144 @@
+{        
+    //Previous definitions
+    const GeometryType& Geom = this->GetGeometry();
+    const GeometryType::IntegrationPointsArrayType& IntegrationPoints = Geom.IntegrationPoints(this->GetIntegrationMethod());
+    const unsigned int NumGPoints = IntegrationPoints.size();
+    const unsigned int LocalDim = Geom.LocalSpaceDimension();
+    
+    //Containers of variables at all integration points
+    const Matrix& NContainer = Geom.ShapeFunctionsValues(this->GetIntegrationMethod());
+    GeometryType::JacobiansType JContainer(NumGPoints);
+    for(unsigned int i = 0; i < NumGPoints; ++i)
+        (JContainer[i]).resize(TDim,LocalDim,false);
+    Geom.Jacobian( JContainer, this->GetIntegrationMethod());
+    
+    //Condition variables
+    array_1d<double,TNumNodes> NormalFluxVector;
+    for(unsigned int i = 0; i < TNumNodes; ++i)
+    {
+        NormalFluxVector[i] = Geom[i].FastGetSolutionStepValue(NORMAL_HEAT_FLUX);
+    }
+    NormalFluxVariables Variables;
+    
+    //Loop over integration points
+    for(unsigned int GPoint = 0; GPoint < NumGPoints; ++GPoint) {
+        //Compute normal flux 
+        Variables.NormalFlux = 0.0;
+        for (unsigned int i = 0; i < TNumNodes; ++i) {
+            Variables.NormalFlux += NContainer(GPoint,i) * NormalFluxVector[i];
+        }
+        
+        //Obtain Np
+        noalias(Variables.Np) = row(NContainer,GPoint);
+                
+        //Compute weighting coefficient for integration
+        this->CalculateIntegrationCoefficient(Variables.IntegrationCoefficient,
+                                              JContainer[GPoint],
+                                              IntegrationPoints[GPoint].Weight() );
+                
+        //Contributions to the right hand side
+        this->CalculateAndAddRHS(rRightHandSideVector, Variables);
+    }
+}
+
+// ============================================================================================
+// ============================================================================================
+template<unsigned int TDim, unsigned int TNumNodes>
+void TNormalFluxCondition<TDim,TNumNodes>::CalculateAndAddRHS(
+    VectorType& rRightHandSideVector,
+    NormalFluxVariables& rVariables )
+{
+    noalias(rVariables.TVector) = rVariables.NormalFlux * rVariables.Np * rVariables.IntegrationCoefficient;
+
+    GeoElementUtilities::
+        AssemblePBlockVector<0, TNumNodes>(rRightHandSideVector, rVariables.TVector);
+}
+
+// ============================================================================================
+// ============================================================================================
+template<unsigned int TDim, unsigned int TNumNodes>
+void TNormalFluxCondition<TDim,TNumNodes>::CalculateIntegrationCoefficient(
+    double& rIntegrationCoefficient,
+    const Matrix& Jacobian,
+    const double& Weight)
+{
+    Vector NormalVector = ZeroVector(TDim);
+
+    if (TDim == 2)
```

Comment:
you can use the newly added `if constexpr (TDim == 2){` which checks the if at compile time --> faster